netplay/joinroom.cpp

```diff
@@ -208,22 +210,64 @@ void JoinRoom::joinGame()
     }
 }
 
-void JoinRoom::serverChanged(QString)
+void JoinRoom::serverChanged(QString serverName)
 {
     if (webSocket)
     {
         webSocket->close();
         webSocket->deleteLater();
+        webSocket = NULL;
+    }
+
+    if (serverName == "Custom")
+    {
+        bool ok;
+        customServerAddress = QInputDialog::getText(0, "Custom Netplay Server", "IP Address / Host:", QLineEdit::Normal, "", &ok);
+
+        if (!ok || customServerAddress.isEmpty())
+        {
+            customServerAddress = QString::null;
+            resetList();
+            return;
+        }
+        else
+        {
+            customServerAddress.prepend("ws://");
+        }
+    }
+    else
+    {
+        customServerAddress = QString::null;
     }
 
     resetList();
     webSocket = new QWebSocket();
     connect(webSocket, &QWebSocket::connected, this, &JoinRoom::onConnected);
-    webSocket->open(QUrl(serverChooser->currentData().toString()));
+    connectionTimer = new QTimer(this);
```

Comment:
Same comment about the timer here. It's better to connect a slot to the error signal of the webSocket if you want to run code when the connection fails

Reply:
The error signal wasn't working for me - it wouldn't ever fire when I put in an IP address that was not running m64p-netplay-server. I can try this again, though, to see if there was something else going on that was causing that to happen.